src/devices/Arduino/ArduinoI2cDevice.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.I2c;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Iot.Device.Arduino
+{
+    /// <summary>
+    /// Implementation of an I2C device connected to an arduino using the Firmata protocol
+    /// </summary>
+    internal class ArduinoI2cDevice : I2cDevice
+    {
+        private readonly ArduinoBoard _board;
+
+        public ArduinoI2cDevice(ArduinoBoard board, I2cConnectionSettings connectionSettings)
+        {
+            _board = board ?? throw new ArgumentNullException(nameof(board));
```

Comment:
no need to do all validation considering this is internal class. Just make sure you never pass invalid values. Note this is optional feedback - more validation doesn't hurt.